sql/core/src/main/scala/org/apache/spark/sql/streaming/StreamingQueryManager.scala

```diff
@@ -265,31 +278,53 @@ class StreamingQueryManager private[sql] (sparkSession: SparkSession) {
       }
```

Comment:
Moving all the code to prepare and create the query object to a separate function would make this function smaller and easier to read.